### PR TITLE
fix: APIs should not be tagged as collection

### PR DESCRIPTION
**Pull request by tuxology**
* use updated joern which populates fullname in annotation
 * skip retrofit annotations when tagging collections

### CODE REVIEW

File: src/test/scala/ai/privado/languageEngine/kotlin/tagger/sink/framework/retrofit/RetrofitAPITaggerTest.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import ai.privado.cache.RuleCache
+import ai.privado.model.*
+import ai.privado.rule.RuleInfoTestData
+import ai.privado.tagger.sink.api.APIValidator
+import ai.privado.entrypoint.PrivadoInput
+import ai.privado.model.{CatLevelOne, Constants, SourceCodeModel}
+
+class RetrofitAPITaggerTest extends KotlinFrontendTestSuite {
+
+  val collectionRule = List(
```
Comment (karan-batavia) [inline src/test/scala/ai/privado/languageEngine/kotlin/tagger/sink/framework/retrofit/RetrofitAPITaggerTest.scala:14]: Is there a base class where these rules could be added? If these are not specific to this test, can we move them there?

**Review by karan-batavia — CHANGES_REQUESTED** [review]

File: src/test/scala/ai/privado/rule/CollectionRuleTestData.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package ai.privado.rule
+
+import ai.privado.model.{CatLevelOne, Constants, FilterProperty, Language, NodeType, RuleInfo}
+
+object CollectionRuleTestData {
```
Comment (karan-batavia) [inline src/test/scala/ai/privado/rule/CollectionRuleTestData.scala:5]: Shouldn't this be under the package `ai.privado.rule.kotlin`? Small change but will ensure better consistency, as this rule is only for kotlin.

**Review by karan-batavia — CHANGES_REQUESTED** [review]

**Comment by karan-batavia** [comment]
One small comment, rest looks good. Also, I see missing flows for python? Can we verify those before merging?

File: src/test/scala/ai/privado/rule/CollectionRuleTestData.scala (hunk 2 above)
Comment (tuxology) [reply src/test/scala/ai/privado/rule/CollectionRuleTestData.scala:5]: Ideally this should work for any language. I'll redefine the rule to make it work for any language so we don't need to move it to any any other package

**Review by karan-batavia — APPROVED** [review]
LGTM

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Pull request merged**